learntools/machine_learning/nbs/tut5.ipynb

```diff
@@ -0,0 +1,186 @@
+    "    preds_val = model.predict(predictors_val)\n",
+    "    mae = mean_absolute_error(targ_val, preds_val)\n",
+    "    return(mae)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "_cell_guid": "eb3819fb-c22c-40e1-8f9f-41d418aaff0f",
+    "_uuid": "57855f1ceb2d184b62aac608b1acf27cf5f4ce45"
+   },
+   "source": [
+    "The data is loaded into **train_X**, **val_X**, **train_y** and **val_y** using the code you've already seen (and which you've already written)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "_cell_guid": "c4156525-eda9-4276-8748-bb7ba1a70745",
+    "_kg_hide-input": true,
+    "_kg_hide-output": true,
+    "_uuid": "3f660cad6e3937c5a6381de48399053fd6d32486",
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# Data Loading Code Runs At This Point\n",
+    "import pandas as pd\n",
+    "    \n",
+    "# Load data\n",
+    "melbourne_file_path = '../input/melbourne-housing-snapshot/melb_data.csv'\n",
+    "melbourne_data = pd.read_csv(melbourne_file_path) \n",
+    "# Filter rows with missing values\n",
+    "filtered_melbourne_data = melbourne_data.dropna(axis=0)\n",
+    "# Choose target and predictors\n",
+    "y = filtered_melbourne_data.Price\n",
+    "melbourne_predictors = ['Rooms', 'Bathroom', 'Landsize', 'BuildingArea', \n",
+    "                        'YearBuilt', 'Lattitude', 'Longtitude']\n",
+    "X = filtered_melbourne_data[melbourne_predictors]\n",
+    "\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "\n",
+    "# split data into training and validation data, for both predictors and target\n",
+    "train_X, val_X, train_y, val_y = train_test_split(X, y,random_state = 0)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "_cell_guid": "0737e6f9-1c82-49bd-b029-103fba702d35",
+    "_uuid": "a519bacabb724ed24b77e24680ac2263a74b63a7"
+   },
+   "source": [
+    "We can use a for-loop to compare the accuracy of models built with different values for *max_leaf_nodes.*"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "_cell_guid": "91c024da-525f-496d-90a5-ac395dd5b87b",
+    "_uuid": "6751d138eff1d9eca94146995ce84554132177d5",
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# compare MAE with differing values of max_leaf_nodes\n",
+    "for max_leaf_nodes in [5, 50, 500, 5000]:\n",
+    "    my_mae = get_mae(max_leaf_nodes, train_X, val_X, train_y, val_y)\n",
+    "    print(\"Max leaf nodes: %d  \\t\\t Mean Absolute Error:  %d\" %(max_leaf_nodes, my_mae))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "_cell_guid": "5f2eb504-a4e5-4857-a974-6830ff34c042",
+    "_uuid": "98c63957ec7c7c7f235e659ca5e6a2aea6e36ad3"
+   },
+   "source": [
+    "\n",
+    "Of the options listed, 500 is the optimal number of leaves.  Apply the function to your Iowa data to find the best decision tree.\n",
```

Comment:
Last sentence is kind of funny. Should it be something like "In the next exercise, you will..."?